src/Network/DogStatsd.hs

```diff
@@ -0,0 +1,53 @@
+module Network.DogStatsd (
+ dogStatsdClient
+, increment
+, decrement
+, count
+, gauge
+, timing
+, histogram
+) where
+
+import Network.Statsd (Stat, Type(..), fmtDatagram)
+import Network.Statsd.UdpClient(UdpClient(..), fromURI, send)
+
+import Control.Monad
+import Data.Maybe
+import Data.List
+import Data.Time.Units
+import Text.Printf
+import Network.URI
+
+type Tags = [(String, String)]
+
+dogStatsdClient :: String -> IO UdpClient
+dogStatsdClient url = (fromURI . fromJust . parseURI) url
+
+increment :: UdpClient -> Stat -> Tags -> IO ()
+increment client stat = count client stat 1
+
+decrement :: UdpClient -> Stat -> Tags -> IO ()
+decrement client stat = count client stat (-1)
+
+count :: UdpClient -> Stat -> Int -> Tags -> IO ()
+count client stat value tags = void . send client $ fmtDogStatsdDatagram stat value Count tags
+
+gauge :: UdpClient -> Stat -> Int -> Tags -> IO ()
+gauge client stat value tags = void . send client $ fmtDogStatsdDatagram stat value Gauge tags
+
+timing :: UdpClient -> Stat -> Millisecond -> Tags -> IO ()
+timing client stat value tags = void . send client $ fmtDogStatsdDatagram stat (fromIntegral value) Timing tags
+
+histogram :: UdpClient -> Stat -> Int -> Tags -> IO ()
+histogram client stat value tags = void . send client $ fmtDogStatsdDatagram stat value Histogram tags
+
+fmtTag :: (String, String) -> String
```

Comment:
I like it. Added in 5a572c1797fc177fc791184482de7c5a1543a9c4 (went with `Name` instead of `Key` to hint that you can just supply a tag name and no value).